src/sections/Meshery/Meshery-platforms/index.js

```diff
@@ -174,14 +174,28 @@ const supported_platforms = [
 
 const MesheryPlatforms = () => {
   const [currentPlatform, setCurrentPlatform] = useState({});
+  const [installationStepsHeight,setInstallationStepsHeight] = useState(currentPlatform.name ? "200px" : 0);
 
-  const changeCurrentPlatform = (index) => {
-    if (currentPlatform.name && currentPlatform.name === supported_platforms[index].name)
+  const hasSelectedSamePlatform = (index) => currentPlatform.name === supported_platforms[index].name;
+
+  const changeCurrentPlatformState = (index) => {
+    if (currentPlatform.name && hasSelectedSamePlatform(index))
       setCurrentPlatform({});
     else
       setCurrentPlatform(supported_platforms[index]);
+
   };
 
+  const changeCurrentPlatform = (index) => {
+    if(currentPlatform.name && !hasSelectedSamePlatform(index)) {
+      changeCurrentPlatformState(index);
+      return;
+    }
+    setTimeout(() => changeCurrentPlatformState(index), 500);
```

Comment:
Why do we need this setTimeout though? We have already added a delay in the CSS below.

Reply:
That's because, without that timeout, the container would be unmounted instantly, so the transition will not take place

Reply:
Instead of timeout use `useRef` to add a class for expansion and remove it otherwise. `setTimeout` is not the best choice here. Then, you might even need animation for the same or if the transition property works, then that's fine as well.

Reply:
Actually, the issue is that, we are conditionally rendering the container right, so, without `setTimeout`, the desired outcome cannot be achieved ig. For instance, lets say i the user wants to close the container, he will click the button, now, when we use `ref` to add/remove class to it, then, we can't get the animation to work because, the moment he presses the button, the container will be unmounted and the `ref.current` will point to `undefined`

Reply:
Ack.... let's keep the setTimeout for the time being. Since, this is restricted to a very small portion of the site it is not much of a problem.